openshift/Makefile

```diff
@@ -0,0 +1,28 @@
+.PHONY: all imagestream buildconfig deploy route help 
+.DEFAULT_GOAL := help
+
+MKFILE_PATH := $(abspath $(lastword $(MAKEFILE_LIST)))
+THISDIR_PATH := $(patsubst %/,%,$(abspath $(dir $(MKFILE_PATH))))
+
+all: imagestream buildconfig deploy route  
+
+imagestream: ## Create the Imagestreams (APIcast and Builder APICast).
+	oc create -f $(THISDIR_PATH)/01-apicast-cloud-hosted-imagestream.yml
+	oc create -f $(THISDIR_PATH)/02-apicast-builder-imagestream.yml
+
+buildconfig: ## Create the BuildConfig.
+	oc create -f $(THISDIR_PATH)/03-build-config.yml
+
+deploy: ## Create the BuildConfig. Parameters: RELEASE_REF, ENVIRONMENT, CACHE_TTL.   
+	oc new-app -f $(THISDIR_PATH)/04-deployment-template.yml \
+		-p RELEASE_REF=${RELEASE_REF} \
+		-p ENVIRONMENT=${ENVIRONMENT} \
+		-p CACHE_TTL=${CACHE_TTL} 
```

Comment:
Will this crash when `CACHE_TTL` is empty? It should.

Reply:
yes it crashes because there is no predefined value in the template.

Actually it brakes without `RELEASE_REF` AND `ENVIRONMENT` also.

The actuall error message:

```bash
oc new-app -f /Users/dcesario/src/apicast-cloud-hosted/openshift/04-deployment-template.yml \
		-p RELEASE_REF=test-dani3 \
		-p ENVIRONMENT=staging \
		-p CACHE_TTL=
error: error processing template "apicast-staging/apicast-cloud-hosted-deployment": Template "apicast-cloud-hosted-deployment" is invalid: template.parameters[2]: Required value: template.parameters[2]: parameter CACHE_TTL is required and must be specified
make: *** [deploy] Error 1

Reply:
👍